external-crates/move/crates/move-ir-compiler/src/lib.rs

```diff
@@ -9,21 +9,38 @@ pub mod util;
 #[cfg(test)]
 mod unit_tests;
 
+use std::collections::BTreeMap;
+
 use anyhow::Result;
 use move_binary_format::file_format::CompiledModule;
 use move_bytecode_source_map::source_map::SourceMap;
-use move_ir_to_bytecode::{compiler::compile_module, parser::parse_module};
+use move_core_types::account_address::AccountAddress;
+use move_ir_to_bytecode::{compiler::compile_module, parser::parse_module_with_named_addresses};
 
 /// An API for the compiler. Supports setting custom options.
 #[derive(Clone, Debug)]
 pub struct Compiler<'a> {
     /// Extra dependencies to compile with.
     pub deps: Vec<&'a CompiledModule>,
+    pub named_addresses: BTreeMap<String, AccountAddress>,
 }
 
 impl<'a> Compiler<'a> {
     pub fn new(deps: Vec<&'a CompiledModule>) -> Self {
-        Self { deps }
+        Self {
+            deps,
+            named_addresses: BTreeMap::new(),
+        }
+    }
+
+    pub fn new_with_addresses(
+        deps: Vec<&'a CompiledModule>,
+        named_addresses: BTreeMap<String, AccountAddress>,
+    ) -> Self {
```

Comment:
I feel like this should be a setter vs an alternative `new`. Kind of like a flag/builder